### PR TITLE
PSQLODBC Datatype Test for sys.uniqueidentifier

### PR DESCRIPTION
**Pull request by ColinKYuen**
### Description

Adds ODBC Datatype Test for `sys.uniqueidentifier `

Task: BABELFISH-569
Signed-off-by: Colin Yuen [yuenhcol@amazon.com](mailto:yuenhcol@amazon.com)

#### Extra Information Test Cases 
- Create Table
- Insert valid values
- Insert invalid values
- Inserting with function
- Update table with valid values
- Update table with invalid values
- Comparison Operators
- Table View
- Unique Constraint
- Composite Key

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/odbc/psqlodbc/unique_identifier.cpp [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+  const vector<string> NAME_EXPECTED = {"int4", "unknown"};
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  testCommonColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS.size(), COL1_NAME, LENGTH_EXPECTED, 
+    PRECISION_EXPECTED, SCALE_EXPECTED, NAME_EXPECTED);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_UniqueIdentifier, Insertion_Success) {
+  char data[BUFFER_SIZE];
```
Comment (ghost) [inline test/odbc/psqlodbc/unique_identifier.cpp]: Delete unused variable.

File: test/odbc/psqlodbc/unique_identifier.cpp [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+
+  const vector<string> INSERTED_VALUES = {
+    "00000000-0000-0000-0000-000000000000", // Min
+    "0E984725-C51C-4BF4-9960-E1C80E27ABA0", // Rand
+    "FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF"  // Max
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS, tableConstraints);
+  testPrimaryKeys(ServerType::PSQL, SCHEMA_NAME, PKTABLE_NAME, PK_COLUMNS);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
```
Comment (ghost) [inline test/odbc/psqlodbc/unique_identifier.cpp:231]: What about testing insertion failure (i.e. can't have duplicate primary keys)?

File: test/odbc/psqlodbc/unique_identifier.cpp [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
```
Comment (ghost) [inline test/odbc/psqlodbc/unique_identifier.cpp:259]: What about testing insertion failure (i.e. can't have duplicate primary keys)?

**Comment by ghost** [comment]
Could you also update your description for the PR, it says you're testing `sys.money`.

**Review by faviansamatha — APPROVED** [review]

**Review by leszekamz — APPROVED** [review]

**Pull request merged**